### ISSUE TITLE
CPU and RAM usage may increase with code additions

### ISSUE THREAD

**Issue by fourtf** [Chatterino/chatterino2#2732]
**Describe your issue**
Resource usage of Chtterino may grow with new additions. A lot of these additions may pile up to make it uncomfortable to use Chtterino. This may be due to RAM/CPU/Network usage.

**Comment by fourtf** [comment]
I suggest adding benchmarks, that are executed on a daily basis and to display the results as graphs on a website somewhere. That way you can see performance regressions and investigate them.

**Comment by leon-richardt** [comment]
Good idea. The biggest challenge I see with this is the design of representative benchmarks that actually measure performance during normal usage. Since we want to spot regressions introduced by new code additions, we would also need guidelines on how to write representative benchmarks for new features.
